### ISSUE TITLE
Index full place name from geonames

### ISSUE THREAD

**Issue by jcoyne** [samvera/hyrax#1065]
When I choose "Sheffield, South Yorkshire, United Kingdom" and it's indexed as 'Sheffield' ... and ditto for "Sheffield, Tasmania, Australia" 
<img width="609" alt="screen shot 2017-05-26 at 10 24 16 am" src="https://cloud.githubusercontent.com/assets/92044/26500940/8c4f367a-41fd-11e7-9de4-a2850f376ff4.png">

**Comment by no-reply** [comment]
Was this closed in #1956?

**Comment by rjkati** [comment]
This issue is still present in Hyrax 3.0.2. When I add location metadata via geonames, only the first part of the name appears.

Example: https://nurax-dev.curationexperts.com/concern/generic_works/6395w720f?locale=en
The Location metadata should be "Chapel Hill, North Carolina, United States" and "Durham, North Carolina, United States". However, the work show page displays "Chapel Hill" and "Durham".
![geonames](https://user-images.githubusercontent.com/32644254/118538687-84358780-b71c-11eb-88a8-8ea812b818e2.PNG)

**Comment by jeremyf** [comment]
This work was in the `valkyrie` branch but never merged into `main`.  PR #1956 includes the SHA 79a19df136bd85ce2b0f283718fba5bf79c6303d.  When I check which branches contains the SHA I get the following:

```sh
❯ git branch --contains 79a19df136bd85ce2b0f283718fba5bf79c6303d
error: no such commit 79a19df136bd85ce2b0f283718fba5bf79c6303d
```

So, yes, the PR resolved it, but we never got this into `main`.  I'm exploring now.

**Comment by jeremyf** [comment]
I'm reviewing the current implementation, and it varies significantly from the non-merged commit (aka 79a19df136bd85ce2b0f283718fba5bf79c6303d).

I am updating the existing spec to highlight that the `gn:countryCode` node is not used.  Here's the spec:

https://github.com/samvera/hyrax/blob/05d576b43572511a62d00bcf14c9ae330d047fa0/spec/indexers/hyrax/generic_work_indexer_spec.rb#L101-L125

When I look at the https://www.geonames.org/5037649/about.rdf URL, there's no state code, just country code.  Which we know to be inadequate.  However, changes introduced in #1956 are breaking changes (updating the generator and removing constants), so I can't use them directly.  I'll be bringing this up during stand-up.

For now, I'm going to work on getting a local test suite up and running so I can verify what I already have.
